csharp/src/Ice/Internal/ConnectionFactory.cs

```diff
@@ -110,15 +110,10 @@ public void waitUntilFinished()
         }
     }
 
-    public void create(EndpointI[] endpts, bool hasMore, Ice.EndpointSelectionType selType,
+    public void create(List<EndpointI> endpoints, bool hasMore, Ice.EndpointSelectionType selType,
```

Comment:
Why did you change the type from array to list? Should we use IList instead?

Reply:
In the existing code, we "applied overrides" on the array which returned a new `List<EndpointI>`, and then passed this endpoint list to various APIs that expect an `List<EndpointI>`,  not an `IList<EndpointI>`. I didn't try to optimize this code.